src/prommis/uky/tests/test_uky_flowsheet.py

```diff
@@ -118,389 +130,404 @@ def test_build_flowsheet(self, model):
         assert isinstance(model.fs.sep1_solid, Arc)
         assert isinstance(model.fs.sep1_retained_liquid, Arc)
         assert isinstance(model.fs.sep1_liquid, Arc)
-        assert isinstance(model.fs.recycle1, Arc)
-        assert isinstance(model.fs.purge1, Arc)
-        assert isinstance(model.fs.recycle_feed, Arc)
+        assert isinstance(model.fs.mixed_aq_feed, Arc)
+        assert isinstance(model.fs.org_feed, Arc)
+        assert isinstance(model.fs.mixed_org_feed, Arc)
         assert isinstance(model.fs.s03, Arc)
+        assert isinstance(model.fs.load_recycle, Arc)
         assert isinstance(model.fs.s04, Arc)
         assert isinstance(model.fs.s05, Arc)
         assert isinstance(model.fs.s06, Arc)
+        assert isinstance(model.fs.scrub_recycle, Arc)
         assert isinstance(model.fs.s07, Arc)
         assert isinstance(model.fs.s08, Arc)
+        assert isinstance(model.fs.s09, Arc)
+        assert isinstance(model.fs.s10, Arc)
+        assert isinstance(model.fs.s11, Arc)
+        assert isinstance(model.fs.s12, Arc)
+        assert isinstance(model.fs.s13, Arc)
+        assert isinstance(model.fs.org_feed2, Arc)
+        assert isinstance(model.fs.s14, Arc)
+        assert isinstance(model.fs.s15, Arc)
+        assert isinstance(model.fs.s16, Arc)
+        assert isinstance(model.fs.s17, Arc)
+        assert isinstance(model.fs.s18, Arc)
+        assert isinstance(model.fs.s19, Arc)
+        assert isinstance(model.fs.s20, Arc)
+        assert isinstance(model.fs.s21, Arc)
+        assert isinstance(model.fs.s22, Arc)
+        assert isinstance(model.fs.s23, Arc)
         assert isinstance(model.fs.sep2_solid, Arc)
         assert isinstance(model.fs.sep2_recovered_liquid, Arc)
-        assert isinstance(model.fs.purge2, Arc)
-        assert isinstance(model.fs.recycle2, Arc)
+        assert isinstance(model.fs.s24, Arc)
+        assert isinstance(model.fs.s25, Arc)
 
         assert degrees_of_freedom(model) == 0
 
-    @pytest.mark.unit
-    def test_set_dof(self, model):
+    @pytest.mark.component
+    @pytest.mark.solver
+    def test_solve(self, model):
         set_scaling(model)
-        set_operating_conditions(model)
-
-        assert degrees_of_freedom(model) == 0
+        # Create a scaled version of the model to solve
+        scaling = TransformationFactory("core.scale_model")
+        scaled_model = scaling.create_using(model, rename=False)
 
-    @pytest.mark.unit
-    def test_initialize_flowsheet(self, model):
-        initialize_system(model)
+        # Initialize system
+        initialize_system(scaled_model)
 
-        assert degrees_of_freedom(model) == 0
+        # Solve scaled model
+        solve(scaled_model)
 
-    @pytest.mark.integration
-    def test_unit_consistency(self, model):
-        assert_units_consistent(model)
+        # Propagate results back to unscaled model
+        scaling.propagate_solution(scaled_model, model)
```

Comment:
1.) I've added `assert_optimal_termination` to this test

2.) I've based this test on what was done in `test_solve` of the `test_leach_flowsheet`, so I figured it'd be best to have the flowsheet tests be as consistent as possible. That being said, I could separate these out if we think that is better, although it might get a bit complicated with the scaling and propagation.

Reply:
They should be separated if possible, and if the leaching flowsheet is still like this after I reworked it then an issue should be opened to improve it.

Reply:
In one of the recent commits, I've used the main() function as the fixture in the test file (which is done in some WaterTAP flowsheet tests), although this still does not address the "At the moment, the test fails if any step fails, but does not give you much information on where it failed". I've made a few attempts to separate the steps out, but it gets tricky with the scaling and propagation. I think we can update this (and rework the leaching flowsheet test later)